content/architecture/vault.md

```diff
@@ -18,13 +18,223 @@ toc: true
+from the Google documentation.
+
+1.  Navigate via browser to the [Project
+Selector](https://console.cloud.google.com/projectselector2/home/dashboard)
+page. and choose your Google Cloud Platform project.
+
+2.  [Create a DNS
+zone](https://console.cloud.google.com/networking/dns/zones/~new)
+
+1.  Choose Public as your *Zone Type*.
+
+2.  Type a *Zone Name* for your zone.
+
+3.  Input a DNS suffix in *DNS name*, for example
+`acmecorp.example`.
+
+4.  Choose your *DNSSEC* or DNS Security state, which should be set
+to `Off` for this configuration.
+
+5.  (Optional) Input a *Description* for your DNS zone.
+
+6.  Click `Create`.
+
+Once created, the *Zone Details* page loads. *NS* (Name server) and
+*SOA* (Start of autority) records are automatically created for your
+domain (for example `acmecorp.example`)
+
+
+### Configuring External DNS in Jenkins X
+
+Once you have configured Google Cloud DNS, you can use browse the
+[Zones](https://console.cloud.google.com/net-services/dns/zones) page in
+your Google Cloud Platform project to setup your external domain.
+
+ NOTE: External DNS will automatically updates DNS records if you reuse the
+ domain name, so if you delete an old cluster and create a new one it
+ will preserve the same domain configuration for the new cluster.
+
+1.  Choose a unique DNS name; you can use nested domains (for example,
+ `cluster1.acmecorp.example`). Enter the name in the `DNS Name`
+field
+
+2.  Run the `jx create domain` command against your domain name:
+
+```
+jx create domain gke --domain cluster1.acmecorp.example
+```
+
+  1.  The program prompts you to choose your Google Cloud Platform
+project from the available list.
+
+  2.  The program prompts you to update your existing managed servers
+to use the displayed list of Cloud DNS nameservers. Copy the
+list for use in the next steps.
+
+  3.  From the Google Cloud Platform
+[Zones](https://console.cloud.google.com/net-services/dns/zones)
+page, change the *Resource Record Type* to `NS`) and use the default
+values for your domain for for *TTL* (`5`) and *TTL Unit*
+(`minutes`).
+
+  4.  Add the first nameserver to the *Name server* field
+
+  5.  Click `Add item` and add any subsequent nameservers.
+
+  6.  Click `Create`.
+
+  7.  Configure Jenkins X for the new domain names:
+
+    - Edit the `jx-requirements.yaml` file and update the `domain`
+field (in `ingress`) to your domain name, for example
+`cluster1.acmecorp.example`
+
+    - In the *tls* setting, enable TLS with `enabled: true`
+
+    - The resulting `jx-requirements.yaml` entries for these settings
+should look similar to the following:
+
+```yaml
+gitops: true
+ingress:
+  domain: cluster1.acmecorp.example
+  externalDNS: true
+  namespaceSubDomain: -jx.
+  tls:
+email: certifiable@acmecorp.example
+enabled: true
+production: true
+secretStorage: vault
+```
+
+Run `jx boot` for the changes to take effect in your
+environment.
+
+## Creating a Vault
```

Comment:
I wondering if we should move this section after `Using Vault on the CLI` because `boot` already creates a vault and for most users is more important to know how to use it form CLI after the initial setup. Essentially, how they can update secrets/get secrets from the vault created by boot.

`Create a Vault` is nice to have in case some users want to create additional vaults.

Reply:
moved using vault up and updated headers a bit as well. I think we should ideally split the content into "using jx" and "managing jx"